liquibase-standard/src/main/java/liquibase/telemetry/configuration/SegmentTelemetryConfiguration.java

```diff
@@ -1,20 +1,31 @@
 package liquibase.telemetry.configuration;
 
+import liquibase.Scope;
 import liquibase.util.Cache;
 import lombok.Data;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.util.concurrent.atomic.AtomicReference;
 
 @Data
 public class SegmentTelemetryConfiguration implements TelemetryConfiguration {
     private final static Cache<RemoteTelemetryConfiguration> remoteTelemetryConfiguration = new Cache<>(() -> {
-        // todo this needs to be fetched from the config endpoint
-        // todo this should only wait a certain amount of time before it too gives up
-        return new RemoteTelemetryConfiguration(
-                1500000000,
-                "https://api.segment.io/v1/batch",
-                true,
-                false,
-                TelemetryArgs.WRITE_KEY.getCurrentValue()
-        );
+        String url = TelemetryArgs.CONFIG_ENDPOINT_URL.getCurrentValue();
+        AtomicReference<RemoteTelemetryConfiguration> remoteTelemetryConfiguration = new AtomicReference<>();
+        Thread thread = new Thread(() -> {
+            try {
+                InputStream input = new URL(url).openStream();
+                Yaml yaml = new Yaml();
+                remoteTelemetryConfiguration.set(yaml.loadAs(input, RemoteTelemetryConfiguration.class));
+            } catch (Exception e) {
+                Scope.getCurrentScope().getLog(SegmentTelemetryConfiguration.class).fine("Failed to load telemetry configuration from " + url, e);
```

Comment:
When this fails, do we provide some kind of default settings? Or telemetry gets disabled? Or we get a `nullpointer` on `get() ` below?

Reply:
This will return a null pointer on the gets below, but those exceptions are swallowed because all exceptions are swallowed, thus effectively disabling telemetry.